test/misc.jl

```diff
@@ -1386,3 +1386,17 @@ end
     @test (@allocations "a") == 0
     @test (@allocations "a" * "b") == 1
 end
+
+@testset "in_finalizer" begin
+    @test !GC.in_finalizer(Base.current_task())
+    @test !GC.in_finalizer()
+
+    r = Ref(1)
+    in_fin = Ref(false)
+    finalizer(r) do _
+        in_fin[] = GC.in_finalizer(Base.current_task()) &&
+                   GC.in_finalizer()
+    end
+    finalize(r)
```

Comment:
The finalize method does not run in the finalizer context, since it has no restrictions on "stealing" the current Task to do arbitrary work.

Reply:
Makes sense!